services/alerts/src/useAlert.ts

```diff
@@ -0,0 +1,29 @@
+import { useCallback, useContext } from 'react'
+import { AlertsManagerContext } from './AlertsManagerContext'
+import { Alert, AlertOptions } from './types'
+
+export const useAlert = (
+    message: string | React.ReactNode | ((props: any) => string),
+    options: AlertOptions | ((props: any) => AlertOptions) = {}
+) => {
+    const alertsManager = useContext(AlertsManagerContext)
+
+    const show = useCallback(
+        props => {
+            const resolvedMessage =
+                typeof message === 'function' ? message(props) : String(message)
+            const resolvedOptions =
+                typeof options === 'function' ? options(props) : options
+
+            const alert: Alert = {
+                message: resolvedMessage,
+                options: resolvedOptions,
+            }
+
+            alertsManager.add(alert)
+        },
+        [alertsManager, message, options]
+    )
+
+    return show
```

Comment:
As mentioned before, I am now simply returning `show`. It used to return `{ show, hideAll }`,  but I'm not really sure why we would need all of that...

Reply:
I generally against implementing hypothetical use cases. It increases the size of a code base for potentially no reason. We should be able to easily implement it once needed.

+1 for returning an object to prevent breaking changes in the future

Reply:
Yeah, I also would prefer not to deal with a hypothetical use-case right now, and also agree that by returning `{ show }` instead of `show` we can address this issue later on without the need for breaking changes.